internal/commands/remote-list.go

```diff
@@ -129,9 +127,6 @@ func (r *remoteLister) deletions(all []model.K8sLocalObject, filter func(obj mod
 
 	var removals []model.K8sQbecMeta
 	for _, c := range all {
-		if r.unknownTypes[c.GroupVersionKind()] {
```

Comment:
Maybe we can classify the unknownTypes rather than just eliminate them. If we can get hold of the API error(Internal Server Error, timeout etc.) while getting server objects, it might give us more mileage in solving the issue with unexpected deletions while also preserving the efficiency gains that we get by ignoring many errors. Maybe the client can use a new cli switch to toggle the behaviour while running qbec commands. A slightly different not-very-user-friendly way that I can think of is passing a list of api-resources in qbec.yaml that must be resolvable before we could do a certain deployment action in an env.

Reply:
Thanks a lot for your review. When I understand it correctly, you suggest classifying which kind of error happened at https://github.com/splunk/qbec/blob/13d7746613f84e91ec3f85e017e453673519655c/internal/commands/remote-list.go#L69 (Internal Server Error, timeout, etc) and based on this, it will `continue` here or not. Unfortunately, I haven't understood what's the purpose of the `unknownTypes` map yet. As far as I understand, it will always delete things on the remote site when there was an error at the referenced line, no matter which kind of classified error.  

In general, I agree with everything which solves the issue and I try to help as much as possible. Also feel free to modify this PR directly or create an own one when this might be easier/faster for you.